### PR TITLE
COMMERCE-4764 Reprocess action always works regardless of the presence of shipment items

### PR DESCRIPTION
**Pull request by riccardo-alberti**

### CODE REVIEW

**Comment by riccardo-alberti** [comment]
ci:test:sf

**Comment by riccardo-alberti** [comment]
ci:test:relevant

**Review by dsitu — APPROVED** [review]

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceShipmentLocalServiceImpl.java [hunk 1]
```diff
@@ -282,6 +282,27 @@ public int getCommerceShipmentsCount(
 			findCommerceShipmentStatusesByCommerceOrderId(commerceOrderId);
 	}
 
+	@Indexable(type = IndexableType.REINDEX)
+	@Override
+	public CommerceShipment reprocessShipment(long commerceShipmentId)
+		throws PortalException {
+
+		CommerceShipment commerceShipment =
```
Comment (marco-leo) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceShipmentLocalServiceImpl.java:290]: @riccardo-alberti  This means that ther eis no need of validation and from any status to can change back to SHIPMENT_STATUS_PROCESSING?

**Comment by luca-pellizzon** [comment]
Check https://github.com/liferay-commerce/liferay-portal/pull/640

**Pull request closed without merging**